processor/spanmetricsprocessor/config.go

```diff
@@ -78,6 +78,14 @@ type Config struct {
 	// the instrumentation library name from the span.
 	// Optional. The default value is `false` which will define the instrumentation library name on metrics as `spanmetricsprocessor`.
 	InheritInstrumentationLibraryName bool `mapstructure:"inherit_instrumentation_library_name"`
+
```

Comment:
This is a pretty big PR for just adding a feature flag, is it necessary? I thought you would hard code a feature flag ID, put in an if statement and be done with it. Is there a reason it needs to be more complex than that ?

Reply:
^I think we probably don't need config to enable it we can just hardcode it to always work (esp since this is just as our fork).

I'm also not sure if we want to commit the LaunchDarklyKey since that's usually treated like a secret (if I'm understanding correctly and that's where the sdk key for auth goes).

Reply:
@jamesmoessis I've hardcoded the id here: https://github.com/atlassian-forks/opentelemetry-collector-contrib/pull/1415/files#diff-8d76ad59560f73ca015339963ad15768144c0a5bceb4e6c062ee371c8d563d69R37

I'm a bit pessimistic when we can get a functional parity processor from the upstream repo. That's why I feel we should do this in a proper way instead of hard code everything.

@Tenaria the LD SDK key will be passed in from an env_var(like [this](https://bitbucket.org/observability/obsvs-ingestion/src/dcb717c720c6ecd68c381dbe4ab560855435c8c5/config/obsvs-config.yml.in#lines-33)). Yes, it should be treated as a secrete. We will write it into Micros Stash.

Reply:
> the LD SDK key will be passed in from an env_var(like this). Yes, it should be treated as a secrete. We will write it into Micros Stash.
ah yeah true 🤦 

> I'm a bit pessimistic about when we can get a functional parity processor from the upstream repo. That's why I feel we should do this in a proper way instead of hard code everything.
I do think it's a bit overkill/improper to add in config for feature flag as it adds another layer that can be confusing to use. For example, I expect that if I turn off the feature flag that it should disable to feature and vice versa but now there's an extra layer where you have to enable it via config too which can be confusing to use. When we don't need the feature flag anymore, we should entirely remove it from code, not just disable it via config.

I'll approve because I don't think it's a major issue given we'll just hardcode it to always be on in the config in the ingestion service and then eventually move upstream. But I'll leave it up to you to decide whether to simplify it or keep as is.

Reply:
> I do think it's a bit overkill/improper to add in config for feature flag as it adds another layer that can be confusing to use. For example, I expect that if I turn off the feature flag it should disable to feature and vice versa but now there's an extra layer where you have to enable it via config too.

I think you are suggesting removing the configuration stuff but keep the change here https://github.com/atlassian-forks/opentelemetry-collector-contrib/pull/1415/files#diff-78f9e7446e0e4fb474786b4b2c262486887bf27479f1f8bc598376d6c3c29b72R128?
Yeah, I think that's fair enough too. But I don't think we should directly hard code the FF without this injection. Otherwise, the code would be hard to be tested.

We need to add the configuration field `LaunchDarklyKey` anyway. We need to use it for configuring the LD SDK Key from ENV_VAR. In that way I don't think we can just keep  `EnableFeatureFlag ` field as well.

Reply:
Cool sounds like we can use Micros stash, I think that's a good idea.